libarchive/archive_write_set_format_pax.c

```diff
@@ -331,17 +348,42 @@ add_pax_attr_binary(struct archive_string *as, const char *key,
 	archive_strappend_char(as, '\n');
 }
 
+static void
+archive_write_pax_header_xattr(struct pax *pax, const char *encoded_name,
+    const void *value, size_t value_len)
+{
+	struct archive_string s;
+	char *encoded_value;
+
+	if ((pax->flags & PAX_FLAG_GNUTAR) == 0) {
+		encoded_value = base64_encode((const char *)value, value_len);
+
+		if (encoded_name != NULL && encoded_value != NULL) {
+			archive_string_init(&s);
+			archive_strcpy(&s, "LIBARCHIVE.xattr.");
+			archive_strcat(&s, encoded_name);
+			add_pax_attr(&(pax->pax_header), s.s, encoded_value);
+			archive_string_free(&s);
+		}
+		free(encoded_value);
+	} else {
```

Comment:
I am now setting both flags for the 'pax' and 'posix' formats. I updated the command line tests above that show that GNU tar displays an error when it encounters LIBARCHIVE.xattr. In that command line test as well as in the test case (patch 2) we have a binary extended attribute with null bytes (0x00) in the middle.